File: Coding/cpp/TreeNode.cpp
#ifndef TREENODE_H // Todo: Learn how this class is implemented
#define TREENODE_H // Todo: How it gets imported?

#include<iostream>
using namespace std;

class TreeNode {
    public:
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int val): val(val), left(nullptr), right(nullptr) {}
    TreeNode(int val, TreeNode* left, TreeNode* right): left(left), right(right) {}
};

#endif // TREENODE_H

File: Problems/java/TreeNode.java
package Problems.java;

public class TreeNode {
    public TreeNode left = null;
    public TreeNode right = null;
    public int val;

    public TreeNode(int val){
        this.val = val;
    }
    
    public TreeNode(int val, TreeNode left, TreeNode right){
        this.left = left;
        this.right = right;
        this.val = val;
    }
    public String toString(){
        return ""+this.val;
    }
}
